tests/ibeam/src/test_gateway_client.py

```diff
@@ -0,0 +1,220 @@
+"""
+Tests for ibeam.src.gateway_client
+"""
+import os
+import pytest
+import random
+import string
+import types
+
+from unittest import mock
+
+from ibeam.src.gateway_client import SECRETS_SOURCE_ENV
+from ibeam.src.gateway_client import SECRETS_SOURCE_FS
+from ibeam.src.gateway_client import GatewayClient
+
+
+@mock.patch('ibeam.src.gateway_client.input')
+@mock.patch('ibeam.src.gateway_client.getpass')
+def test_init_prompt(mock_input, mock_getpass):
+    """
+    GatewayClient.__init__ test stdout/stdin prompts for
+    secrets.
+    """
+
+    # call fake_input when input and getpass are called
+    def fake_input(s):
+        return 'fake ' + s + 'response'
+    mock_input.side_effect = fake_input
+    mock_getpass.side_effect = fake_input
+
+    # test init w/o any setup, which should write prompts to
+    # stdout and should read stdin for the values
+    client = GatewayClient(
+        http_handler=None, inputs_handler=None, two_fa_handler=None)
+
+    assert client.account == 'fake Account: response'
+    assert client.password == 'fake Password: response'
+    assert client.key == 'fake Key: response'
+
+
+def test_secret_value(tmpdir):
+    """
+    GatewayClient.secret_value tests both via init and when
+    directly called
+    """
+
+    # rnd generates a random value w/o any control
+    # characters
+    def rnd(n=16):
+        c = string.ascii_letters + string.digits + string.punctuation
+        return ''.join(random.choice(c) for i in range(n))
+
+    # basic tests to run include specifying the secrets
+    # directly, reading them from the environment, and
+    # reading them from the filesystem
+    tests = [
+        types.SimpleNamespace(
+            name='passed values',
+            source=None,
+            initial_env={
+            },
+        ),
+        types.SimpleNamespace(
+            name='env values w/o IBEAM_SECRETS_SOURCE',
+            source=SECRETS_SOURCE_ENV,
+            initial_env={
+            },
+        ),
+        types.SimpleNamespace(
+            name='env values w/ IBEAM_SECRETS_SOURCE',
+            source=SECRETS_SOURCE_ENV,
+            initial_env={
+                'IBEAM_SECRETS_SOURCE': SECRETS_SOURCE_ENV,
+            },
+        ),
+        types.SimpleNamespace(
+            name='fs values w/ IBEAM_SECRETS_SOURCE',
+            source=SECRETS_SOURCE_FS,
+            initial_env={
+                'IBEAM_SECRETS_SOURCE': SECRETS_SOURCE_FS,
+            },
+        ),
+    ]
+
+    # names of the secret values
+    ibeam_fields = [
+        'IBEAM_ACCOUNT',
+        'IBEAM_PASSWORD',
+        'IBEAM_KEY',
+    ]
+
+    for t in tests:
+        # GatewayClient init variables
+        account = None
+        password = None
+        key = None
+
+        # generate random secret values
+        for k in ibeam_fields:
+            setattr(t, k, rnd())
```

Comment:
I'd suggest writing out `test` and later `field` variables instead of `t` and `k` for readability.